### PR TITLE
screen-pipe: init at 0.1.48

### LINKED ISSUE #331110: Package request: screen-pipe
**Project description**

Library to build personalized AI powered by what you've seen, said, or heard. Works with Ollama. Alternative to Rewind.ai. Open. Secure. You own your data. Rust. Has both CLI and desktop app.

**Metadata**

* homepage URL: https://screenpi.pe
* source URL: https://github.com/louis030195/screen-pipe
* license: mit
* platforms: unix, linux, darwin, windows

### PR DESCRIPTION
**Pull request by dit7ya**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/331110.

Only built on aarch64-darwin, haven't tested more than running the binaries for a few seconds. Will look into packaging the tauri desktop app, but doesn't look very straight-forward.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sc/screen-pipe/package.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    darwin.apple_sdk_12_3.frameworks.CoreFoundation
+    darwin.apple_sdk_12_3.frameworks.CoreGraphics
+    darwin.apple_sdk_12_3.frameworks.CoreMedia
+    darwin.apple_sdk_12_3.frameworks.IOKit
+    darwin.apple_sdk_12_3.frameworks.Metal
+    darwin.apple_sdk_12_3.frameworks.MetalPerformanceShaders
+    darwin.apple_sdk_12_3.frameworks.Security
+    darwin.apple_sdk_12_3.frameworks.ScreenCaptureKit
+    darwin.apple_sdk_12_3.frameworks.SystemConfiguration
+  ] ++ lib.optionals stdenv.isLinux [
```
Comment (afh) [inline pkgs/by-name/sc/screen-pipe/package.nix]: Using `with` here would make for more readable and maintainable darwin `buildInputs`
```suggestion
  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk_12_3.frameworks; [
    CoreAudio
    AudioUnit
    CoreFoundation
    CoreGraphics
    CoreMedia
    IOKit
    Metal
    MetalPerformanceShaders
    Security
    ScreenCaptureKit
    SystemConfiguration
  ]) ++ lib.optionals stdenv.isLinux [
```

File: pkgs/by-name/sc/screen-pipe/package.nix [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      darwin.apple_sdk_12_3.frameworks.CoreFoundation
+      darwin.apple_sdk_12_3.frameworks.CoreGraphics
+      darwin.apple_sdk_12_3.frameworks.CoreMedia
+      darwin.apple_sdk_12_3.frameworks.IOKit
+      darwin.apple_sdk_12_3.frameworks.Metal
+      darwin.apple_sdk_12_3.frameworks.MetalPerformanceShaders
+      darwin.apple_sdk_12_3.frameworks.Security
+      darwin.apple_sdk_12_3.frameworks.ScreenCaptureKit
+      darwin.apple_sdk_12_3.frameworks.SystemConfiguration
+    ]
```
Comment (afh) [inline pkgs/by-name/sc/screen-pipe/package.nix:65]: Using `with` here would improve readability and maintainability of the darwin `buildInputs`:
```suggestion
    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk_12_3.frameworks; [
      CoreAudio
      AudioUnit
      CoreFoundation
      CoreGraphics
      CoreMedia
      IOKit
      Metal
      MetalPerformanceShaders
      Security
      ScreenCaptureKit
      SystemConfiguration
    ])
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @dit7ya, please find below a suggestion on how to improve the darwin `buildInputs`.

**Comment by afh** [comment]
Result of `nixpkgs-review pr 332076` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>screen-pipe</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for accepting the suggestions, @dit7ya, much appreciated 👍

**Comment by marcusramberg** [comment]
builds ok on x86 linux too.

**Comment by afh** [comment]
Thanks for the contribution, @dit7ya, and for merging, @marcusramberg, very much appreciated 🙂 👍

**Comment by louis030195** [comment]
@dit7ya LMK if you need help with the tauri app, I'm building screen-pipe! 

just DM on louis030195 (discord, x, etc.)

**Comment by louis030195** [comment]
@dit7ya hey btw we moved repo to https://github.com/mediar-ai/screenpipe - hope it's not breaking anything

**Pull request merged**